ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -613,4 +630,32 @@ private[kvutils] object TransactionCommitter {
   def getContractState(commitContext: CommitContext, key: DamlStateKey): DamlContractState =
     commitContext.get(key).getOrElse(throw Err.MissingInputState(key)).getContractState
 
+  @SuppressWarnings(Array("org.wartremover.warts.Option2Iterable"))
+  private def transactionMinRecordTime(
```

Comment:
This property will be read by `KeyValueCommitting.dryRun` for pre-execution (see #6592 ). In case of pre-execution we are delaying the time-bound checks to post-execution as the record time will be only known then. Hence, we have to keep track of the time-bounds that are influenced by submission time and command deduplication time window in case of transactions. In the post-execution phase we only check the current record time against the time-bounds determined by the `Commiter`s.